src/kakarot/accounts/library.cairo

```diff
@@ -250,8 +250,11 @@ namespace AccountContract {
         // The protocol nonce is updated once per __execute__ call, while the EVM nonce is updated once per
         // transaction. If we were to execute more than one transaction in a single __execute__ call, we would
         // need to change the nonce incrementation logic.
-        let (current_nonce) = Account_nonce.read();
-        Account_nonce.write(current_nonce + 1);
+        //
+        // TODO! If the previous execute failed with a CairoVM error, the protocol nonce was not incremented
```

Comment:
how could we have a cairo vm error? we should not have any such error

Reply:
Run resources errors, we do not have a way to handle these for now.